Add support for release handling

tehsmyers opened the pull request:
Specifically, this adds support for the `--set` and `--unset` flags to
the `subscription-manager release` subcommand.

Closes #6

tehsmyers commented:
I'd like to test the hell out of this before merging.

tehsmyers commented:
I don't know why travis is failing, looks like an incompatibility between how molecule (or the underlying ansible module) is trying to call into docker versus the docker api. The downstream tests have not been updated to cover this, but their passing at least means that this change doesn't break other tasks. :)

robled commented on README.md:
```diff
@@ -92,11 +97,11 @@ Role Output
-  This value will be `false` if the `rhsm_repositories` input var is not used.
 - `subscribed` - Whether or not the system is registered. (bool)
-- `subscribed_pool_ids` - A list of pool IDs current attached too the system. Will be an empty list if no pools or attached,
+- `subscribed_pool_ids` - A list of pool IDs current attached too the system. Will be an empty list if no pools are attached,
```
typo - 'too'

robled commented on defaults/main.yml:
```diff
@@ -68,6 +68,13 @@ rhsm_force_register: "{{ omit }}"
 # yes/no bool, default no
 rhsm_unregister: false
 
+# Bool, whether ot not to unset the subscription-manager release,
```
typo - 'ot'

greg-hellings commented on tasks/release.yml:
```diff
@@ -0,0 +1,15 @@
+- name: Unset subscription-manager release
+  command: subscription-manager release --unset
+  when: rhsm_release_unset and oasis_roles_rhsm_release
+  register: rhsm_release_unset_task
```
On this task and the next, is there any way to determine if changes were made? Sometimes there's a way to set `changed_when` based on the stdout of the command or such?

tehsmyers replied:
Sadly, no. Here's the output I'm working with:
```
# subscription-manager release --show
Release: 7.0
# subscription-manager release --set 7.0
Release set to: 7.0
# subscription-manager release --unset
Release preference has been unset
# subscription-manager release --unset
Release preference has been unset
```

So you can see we still get the "Release set to" text, even when the release was already set to the "new" value. The same is true when using `unset`; s-m reports that a change was made, even when that change is to the current value.

robled replied:
I suggest that we embellish the comment a bit to explain the unfortunate behavior here.  Sounds like getting this into the module itself is the real path forward.  Until then we can have some info to explain why we can't track actual changes here.

tehsmyers replied:
I'm not sure what you're asking for. The unfortunate behavior is that we have to set facts internally to ensure idempotence, and the fix is to put the behavior into the rhsm_repository module. That's what the comment already says.

I don't think it's worthwhile to write a 20-line comment in the role to explain this; I was planning to open up a github issue spelling out the problem fully once I actually had code in this repo to link to.

robled replied:
I was asking for info in the code comments about the subscription manager output, similar to your earlier comment in this thread.  As someone who hasn't done much with s-m, this seemed useful to me.  However, a GitHub issue with that info would also suffice.

tehsmyers replied:
Cool. I'll make sure to reference this comment in the issue and see about providing a more complete picture of how s-m is working for us (and against us) here, and how we can get this taken care of upstream.

greg-hellings commented on tasks/release_fact.yml:
```diff
@@ -0,0 +1,21 @@
+  command: subscription-manager release --show
+  register: rhsm_release_check
+  changed_when: false
+  failed_when: false
```
If `failed_when: false` should this also have `ignore_errors: true`? Or do we still want to capture non-zero exit statuses from the command?

tehsmyers replied:
We do still want to capture non-zero exist statuses from the command, it's used on line 21.

greg-hellings commented on tests/yamllint.yml:
```diff
@@ -3,6 +3,7 @@ extends: default
 rules:
   line-length:
     level: error
+    allow-non-breakable-inline-mappings: true
```
What does this do, and is it something we should include in the meta_skeleton for future roles?

tehsmyers replied:
It allows line-length to exceed the default maximum in non-breakable strings in inline mappings. In my case, it lets me put longish urls into a playbook without having to mangle them or do other trickery. This is for a specific downstream test scenario, though, so it's probably worth using only in that scenario. I do not think it's worth including in the meta_skeleton by default, but I do think it's good to know about.

tl;dr I'm going to remove this. :)

Commit:
Add support for release handling

Specifically, this adds support for the `--set` and `--unset` flags to
the `subscription-manager release` subcommand.
